### DISCUSSION TITLE
Querying fields in GraphQL

### DISCUSSION THREAD

**Discussion by akve** [prisma/prisma#14407] [Q&A] [answered]
Hi there! 
I got a project still on Prisma1 and curious if it's supported somehow?.. 

There's a lot of code that queries models using GraphQLResolveInfo - say
```
const user = await prisma.query.user(
    { where: { sessionId } },
    `{
    id
    employee {
      id
      firstName
    }
  }`
  );
```

Is this approach feasible with latest Prisma ? Can't find this possibility anywhere - and our queries are so complicated that the feature is incredibly nice to have.

Thanks!
Alex

**Comment by austincrim** [comment, marked as answer]
Hey @akve 👋,

It isn't possible to use Prisma 2+ this way. You could create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for it.

Here's [our official Prisma 1 upgrade guide](https://www.prisma.io/docs/guides/upgrade-guides/upgrade-from-prisma-1/how-to-upgrade) if you haven't seen it.

**Comment by akve** [comment]
@austincrim thanks! I know it's not possible, and full upgrade doesn't make sense so far - the project is using GraphQL as a language for both "outer world API" and ORM language (there's a lot of complicated things to be calculated). 

So far, found https://github.com/paljs/prisma-tools that can help with creating prisma select from GraphQLResolveInfo, but now can't find an easy way to generate GraphQLResolveInfo from a string ) 

Thanks! 
Alex

**Comment by akve** [comment]
One more possible option for such scenarios would be lifting intermediate GraphQL server that transparently replies to full Prisma2 schema. Is it possible to "forward" all possible requests, matching all the models? Now stuck with this approach :)

@austincrim your opinion matters..
Thanks!

**Reply by austincrim** [reply]
Could you explain this solution a little more? Maybe show example pseudo-code?

**Reply by akve** [reply]
@austincrim 
the idea is pretty close to what Prisma1 is doing.. 
It's another really small GraphQL server that is serving all data-model based queries and mutations, without custom resolvers. Something that "forward" operation was doing in Prisma1. I failed to find a way to provide "all-in-all-out" way so far, but seems like it might be working approach. 

Actually, this full-forward server is a nice-to-have example, anyway. I know it would be schema-first, but really fit for the situations where the database structure is in place, and only needs GraphQL transparent access to it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
